unit_test.py

```diff
@@ -0,0 +1,1371 @@
+# This is file contains a unit test suit which completely test the backend
+# functions of our app.
+# Note:
+# 1.We create some random functions before the test suit to add randomness to
+# our tests.
+# 2.Some later tests use some of the previous test to avoid code repetitions.
```

Comment:
Hmm.. The earlier tests(such as test_create_class) I used in later tests are more like helper functions. They create an object and store it into the database. Repeting the code inside this function will life harder when one tries to change the code of each of them.